lib/facter/util/resolvers/networking/networking.rb

```diff
@@ -65,7 +65,19 @@ def find_valid_binding(bindings)
           end
 
           def ignored_ip_address(addr)
-            addr.empty? || addr.start_with?('127.', '169.254.') || addr.start_with?('fe80') || addr.eql?('::1')
+            return true if addr.empty?
+
+            ip = IPAddr.new(addr)
+            return true if ip.loopback?
+
+            [
+              IPAddr.new('169.254.0.0/16'), # Link-local unicast RFC5735
+              IPAddr.new('fe80::/10')       # Link-local unicast RFC4291
+            ].each do |range|
+              return true if range.include?(ip)
+            end
```

Comment:
Suggest creating frozen constants with the parsed IPAddr at the top:

```ruby
LINK_LOCAL_ADDR = IPAddr.new('169.254.0.0/16').freeze
```

And then check the ip against it:

```ruby
return true if LINK_LOCAL_ADDR.include?(ip)
```

Reply:
Done!